src/cassandra/result.rs

```diff
@@ -85,13 +82,16 @@ impl CassResult {
     pub fn column_count(&self) -> u64 { unsafe { cass_result_column_count(self.0) as u64 } }
 
     /// Gets the column name at index for the specified result.
-    pub fn column_name(&self, index: usize) -> String {
+    pub fn column_name(&self, index: usize) -> Result<&str> {
         unsafe {
             let name = mem::zeroed();
             let name_length = mem::zeroed();
-            cass_result_column_name(self.0, index, name, name_length);
-            let slice = slice::from_raw_parts(name as *const u8, name_length as usize);
-            str::from_utf8(slice).expect("must be utf8").to_owned()
+            cass_result_column_name(self.0, index, name, name_length).to_result(())
```

Comment:
Surely the real solution is to move everything into Rust and stop using Unsafe code? 😈